### PR TITLE
Database auth for all

### PR DESCRIPTION
**Pull request by orangewolf**
this looks like it will apply fairly cleanly but I'm highlighting some areas in comments that should be discussed / decided on.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,3 +1,4 @@
+SCHOLARWORKS_HOST=localhost
```
Comment (orangewolf) [inline .env:1]: is this a sensible default? Its overrode in prod already

File: Gemfile [hunk 2]
```diff
@@ -92,7 +93,6 @@ gem 'solargraph'
 gem 'colorize'
 gem 'rubyzip'
 gem "ruby-vips", "~> 2.0"
-gem 'diffy'
```
Comment (orangewolf) [inline Gemfile:95]: not sure why this is pulled out specifically in the prod branch

File: app/controllers/users/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Users
+  # session controller to redirect the user to shib login path
+  class SessionsController < Devise::SessionsController
+    def new
+      Rails.logger.debug "SessionsController#new: request.referer = #{request.referer}"
+      if Settings.require_shib_user_authn && params[:direct].blank?
```
Comment (orangewolf) [inline app/controllers/users/sessions_controller.rb:8]: this params thing, the sessions new view and the routes are what I added to make this all work.

File: public/robots.txt [hunk 4]
```diff
@@ -1,2 +1 @@
-User-agent: * 
-Disallow: /
\ No newline at end of file
+Sitemap: https://scholarworks.calstate.edu/sitemap.xml
```
[the remark this answers is not in the document]
Comment (dswalker) [reply public/robots.txt:1]: Thank you for this.  This is fixed now on master.

File: lib/hydra/derivatives/runners/pdf_derivatives.rb [hunk 5]
```diff
@@ -3,9 +3,9 @@
 module Hydra
   module Derivatives
     class PdfDerivatives < ImageDerivatives
-      def self.processor_class
-	Processors::PdfProcessor
-      end
+      # def self.processor_class
+      #   Processors::PdfProcessor
+      # end
```
[the remark this answers is not in the document]
Comment (dswalker) [reply lib/hydra/derivatives/runners/pdf_derivatives.rb:8]: This is code from the libvips customization, so I think should be left as-is.

File: config/environments/production.rb [hunk 6]
```diff
@@ -47,7 +47,7 @@
   # config.action_cable.allowed_request_origins = [ 'http://example.com', /http:\/\/example.*/ ]
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
-  config.force_ssl = false
+  config.force_ssl = true
```
[the remark this answers is not in the document]
Comment (dswalker) [reply config/environments/production.rb]: It does, although we need to get demo using SSL anyway.  I think, though, it's easier for me if we leave this as false and then use Apache to force SSL where appropriate.

File: Gemfile (hunk 2 above)
Comment (dswalker) [reply Gemfile:96]: This should have been removed from both master and production.

**Pull request merged**